cmd/umserver/graceful_shutdown.go

```diff
@@ -0,0 +1,49 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"io"
+	"log"
+	"net/http"
+	"sync"
+	"time"
+)
+
+// closingStructure includes all components, which are connected with our application
+type closingStructure struct {
+	components []io.Closer
+}
+
+func gracefulShutdown(timeout time.Duration, wg *sync.WaitGroup, srv *http.Server, closers *closingStructure) error {
```

Comment:
WDYT to change signature to:
```
gracefulShutdown(timeout time.Duration, wg *sync.WaitGroup, srv *http.Server, closers []io.Closer) error
```

Reply:
or
```
gracefulShutdown(timeout time.Duration, wg *sync.WaitGroup, srv *http.Server, closers ...io.Closer) error
```

Reply:
I agree with you. It is a good idea. I fixed it